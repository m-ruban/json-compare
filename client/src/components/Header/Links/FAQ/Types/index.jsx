import React from 'react';
import trl from 'modules/translation';
import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';

const Types = () => (
    <div>
        {trl('FAQ.ComparisonTypes.Answer.text')}
        <ol>
            <li>
                {EQUALITY_TYPE_STRICT} - {trl('FAQ.ComparisonTypes.Answer.noStrict')}
            </li>
            <li>
                {EQUALITY_TYPE_NOT_STRICT} - {trl('FAQ.ComparisonTypes.Answer.strict')}
            </li>
            <li>
                {EQUALITY_TYPE_IGNORE} - {trl('FAQ.ComparisonTypes.Answer.ignore')}
            </li>
        </ol>
    </div>
);

export default Types;
