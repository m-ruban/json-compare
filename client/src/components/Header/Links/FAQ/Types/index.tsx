import React from 'react';

import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';
import trl from 'modules/translation';

const Types = () => (
    <div>
        {trl('FAQ.ComparisonTypes.Answer.text')}
        <ol>
            <li>
                {EQUALITY_TYPE_STRICT} - {trl('FAQ.ComparisonTypes.Answer.noStrict')}
            </li>
            <li>
                {EQUALITY_TYPE_NOT_STRICT} - {trl('FAQ.ComparisonTypes.Answer.strict')}
            </li>
            <li>
                {EQUALITY_TYPE_IGNORE} - {trl('FAQ.ComparisonTypes.Answer.ignore')}
            </li>
        </ol>
    </div>
);

export default Types;
