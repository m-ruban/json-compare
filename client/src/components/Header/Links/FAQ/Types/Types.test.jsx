import React from 'react';
import { render } from 'enzyme';
import Types from 'components/Header/Links/FAQ/Types';

describe('<Types />', () => {
    it('Renders correctly', () => {
        const output = render(<Types />);
        expect(output).toMatchSnapshot();
    });
});
