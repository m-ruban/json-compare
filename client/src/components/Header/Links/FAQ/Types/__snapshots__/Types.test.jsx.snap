// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Types /> Renders correctly 1`] = `
<div>
  This application has three types of comparison:
  <ol>
    <li>
      STRICT - The application is uses not strict equality (13 == "13" is true)
    </li>
    <li>
      NOT_STRICT - The application is uses strict equality (13 == "13" isn't true, but 13 === 13 is true)
    </li>
    <li>
      IGNORE - The application is ignores value of the key
    </li>
  </ol>
</div>
`;
